_extensions/pillar/qvm_prefs.py

```diff
@@ -34,24 +36,21 @@ def __virtual__():
 
 
 def ext_pillar(minion_id, pillar, *args, **kwargs):
-    qc = QubesVmCollection()
-    qc.lock_db_for_reading()
-    qc.load()
-    qc.unlock_db()
-    vm = qc.get_vm_by_name(minion_id)
+    app = qubes.Qubes()
+    vm = app.domains[minion_id]
```

Comment:
This will throw KeyError instead of returning None for invalid domain - needs try/except.